internal/api/errors.go

```diff
@@ -305,7 +305,7 @@ func HandleResponseError(err error, w http.ResponseWriter, r *http.Request) {
 				Message:    "Unexpected failure, please check server logs for more information",
 			}
 
-			if jsonErr := sendJSON(w, http.StatusInternalServerError, httpError); jsonErr != nil {
+			if jsonErr := sendJSON(w, http.StatusInternalServerError, httpError); jsonErr != nil && jsonErr != context.DeadlineExceeded {
```

Comment:
why do we ignore the error if the context deadline is exceeded?

Reply:
This function recursively calls itself to go down deeper in the error stack... If you remove that when the context exceeds, endless recursion 😓 

This function needs to be reworked a bit.